packages/html-reporter/src/utils.ts

```diff
@@ -47,3 +47,21 @@ export function hashStringToInt(str: string) {
     hash = str.charCodeAt(i) + ((hash << 8) - hash);
   return Math.abs(hash % 6);
 }
+
+export function formatUrl(url: string): string;
+export function formatUrl(url: string | undefined): string | undefined;
+export function formatUrl(url: string | undefined): string | undefined {
+  if (!url)
+    return url;
+  try {
+    const parsed = new URL(url, window.location.href);
+    if (parsed.protocol === window.location.protocol && parsed.origin === window.location.origin) {
```

Comment:
origin includes protocol
```suggestion
    if (parsed.origin === window.location.origin) {
```